adapters/openweb/openweb.go

```diff
@@ -0,0 +1,193 @@
+package openweb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type OpenWebAdapter struct {
+	endpoint string
+}
+
+type openwebImpExt struct {
+	OpenWeb openrtb_ext.ExtImpOpenWeb `json:"openweb"`
+}
+
+func (a *OpenWebAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	totalImps := len(request.Imp)
+	errors := make([]error, 0, totalImps)
+	imp2source := make(map[int][]int)
+
+	for i := 0; i < totalImps; i++ {
+
+		sourceId, err := validateImpression(&request.Imp[i])
+
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if _, ok := imp2source[sourceId]; !ok {
+			imp2source[sourceId] = make([]int, 0, totalImps-i)
+		}
+
+		imp2source[sourceId] = append(imp2source[sourceId], i)
+
+	}
+
+	totalReqs := len(imp2source)
+	if 0 == totalReqs {
+		return nil, errors
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqs := make([]*adapters.RequestData, 0, totalReqs)
+
+	imps := request.Imp
+	request.Imp = make([]openrtb2.Imp, 0, len(imps))
+	for sourceId, impIds := range imp2source {
+		request.Imp = request.Imp[:0]
+
+		for i := 0; i < len(impIds); i++ {
+			request.Imp = append(request.Imp, imps[impIds[i]])
+		}
+
+		body, err := json.Marshal(request)
+		if err != nil {
+			errors = append(errors, fmt.Errorf("error while encoding bidRequest, err: %s", err))
+			return nil, errors
+		}
+
+		reqs = append(reqs, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpoint + fmt.Sprintf("?aid=%d", sourceId),
+			Body:    body,
+			Headers: headers,
+		})
+	}
+
+	if 0 == len(reqs) {
```

Comment:
I don't think this needs to be s special case. `return reqs, errors` should work fine for an empty `reqs` slice.